includes/Template_Post_Type.php

```diff
@@ -0,0 +1,93 @@
+	 *
+	 * @var string
+	 */
+	const POST_TYPE_SLUG = 'web-story-template';
+
+	/**
+	 * Registers the post type for story templates.
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		register_post_type(
+			self::POST_TYPE_SLUG,
+			[
+				'labels'       => [
+					'name'                     => _x( 'Templates', 'post type general name', 'web-stories' ),
+					'singular_name'            => _x( 'Template', 'post type singular name', 'web-stories' ),
+					'add_new'                  => _x( 'New', 'story', 'web-stories' ),
+					'add_new_item'             => __( 'Add New Template', 'web-stories' ),
+					'edit_item'                => __( 'Edit Template', 'web-stories' ),
+					'new_item'                 => __( 'New Template', 'web-stories' ),
+					'view_item'                => __( 'View Template', 'web-stories' ),
+					'view_items'               => __( 'View Templates', 'web-stories' ),
+					'search_items'             => __( 'Search Templates', 'web-stories' ),
+					'not_found'                => __( 'No templates found.', 'web-stories' ),
+					'not_found_in_trash'       => __( 'No templates found in Trash.', 'web-stories' ),
+					'all_items'                => __( 'All Templates', 'web-stories' ),
+					'archives'                 => __( 'Template Archives', 'web-stories' ),
+					'attributes'               => __( 'Template Attributes', 'web-stories' ),
+					'insert_into_item'         => __( 'Insert into template', 'web-stories' ),
+					'uploaded_to_this_item'    => __( 'Uploaded to this story', 'web-stories' ),
+					'featured_image'           => __( 'Featured Image', 'web-stories' ),
+					'set_featured_image'       => __( 'Set featured image', 'web-stories' ),
+					'remove_featured_image'    => __( 'Remove featured image', 'web-stories' ),
+					'use_featured_image'       => __( 'Use as featured image', 'web-stories' ),
+					'filter_items_list'        => __( 'Filter stories list', 'web-stories' ),
+					'items_list_navigation'    => __( 'Templates list navigation', 'web-stories' ),
+					'items_list'               => __( 'Templates list', 'web-stories' ),
+					'item_published'           => __( 'Template published.', 'web-stories' ),
+					'item_published_privately' => __( 'Template published privately.', 'web-stories' ),
+					'item_reverted_to_draft'   => __( 'Template reverted to draft.', 'web-stories' ),
+					'item_scheduled'           => __( 'Template scheduled', 'web-stories' ),
+					'item_updated'             => __( 'Template updated.', 'web-stories' ),
+					'menu_name'                => _x( 'Templates', 'admin menu', 'web-stories' ),
+					'name_admin_bar'           => _x( 'Template', 'add new on admin bar', 'web-stories' ),
+				],
+				'supports'     => [
+					'title', // Used for amp-story[title].
+					'author',
+					'excerpt',
+					'thumbnail', // Used for poster images.
+				],
+				'public'       => false,
+				'show_ui'      => false,
+				'show_in_rest' => true,
```

Comment:
Should we mock out a custom controller class here? 
`'rest_controller_class' => Template_Controller::class,`

Reply:
Oh, I thought it was not necessary. But it turns out it might be, for `story_data`/`post_content_filtered`. Somehow missed that.

Reply:
Copy/pasted the relevant parts for now. We could potentially make this more DRY in the future though. Suggestions welcome.

Reply:
I thought we could use this custom controller to add bookmarking and other filter required.

Reply:
Bookmarking is a whole other story post-stable.